### ISSUE TITLE
Add contacts popover to other apps

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1086]
Opened by @jancborchardt at https://github.com/nextcloud/server/issues/4532

Detailed description and (maybe) mockup is needed

**Comment by AndyScherzinger** [comment]
👍 and which actions need to be made available in the popover in which context...

**Comment by jancborchardt** [comment]
For a mockup, basically look at the mobile web interface of the server. :) It should be a list of contacts, with a search box, and relevant actions like mailing, calling (especially useful on smartphones) etc.

**Comment by tobiasKaminsky** [comment]
Action can be:
- phone number: dial directly
- address: open in maps
- email: compose email 
- twitter: open in twitter app or in browser 

But where do I see those contacts? In WebUI it is easy as you can see the sharee in the side menu, but in android this is done already as a popup.

**Comment by jancborchardt** [comment]
Yeah, any place where users are shown. Is it a problem that the share menu is already a popup? Couldn't a click of their name/avatar open a menu popover like on the server?

**Comment by AndyScherzinger** [comment]
Popovers could be implemented, though that'll be some work and we would be stacking quite some UI elements then. The search within a popover seems a bit unorthodox since I can't recall any Android app using popovers for functionality but rather to display information (while a click action should be fine) and not containing any UI elements with which you can "interact" (except for the click of course). Will have to think about it a bit more how this can be done nicely without creating a Windows Mobile 6 like UI... ;)

**Comment by jancborchardt** [comment]
Then maybe we should move away from the popover for sharing? Maybe even model it more like the web app with a sidebar sliding in from the right?

**Comment by AndyScherzinger** [comment]
We should move away from the popover for sharing, definitely! It is actually an activity (full screen UI) which uses a special theme/style to make it look like a popup. A sliding sidebar from the right also would be non-Android style. Like I said I need to think about it a little more. In case we move away from the popup theming it'll get better UI element stacking wise and the contacts popover could then be the popup and provide search, etc. which would also work better in case a lot of content needs to be displayed since the popup can have a scroll functionality :)
